pytorch_lightning/utilities/memory.py

```diff
@@ -75,5 +76,13 @@ def is_out_of_cpu_memory(exception):
 def garbage_collection_cuda():
     """Garbage collection Torch (CUDA) memory."""
     gc.collect()
-    if torch.cuda.is_available():
-        torch.cuda.empty_cache()
+    try:
+        if torch.cuda.is_available():
+            # This is the last thing that should cause an OOM error, but seemingly it can.
+            torch.cuda.empty_cache()
+    except RuntimeError as exception:
+        if is_oom_error(exception):
```

Comment:
can you invert this logic and avoid the else?

Reply:
I'm tempted to add a `raise_otherwise=False` keyword argument to `is_oom_error` to avoid even the `if`. What do you think?

Reply:
I don't think it's necessary

```python
if not oom_error(e):
    raise
```

should be enough